### DISCUSSION TITLE
How to make text image

### DISCUSSION THREAD

**Discussion by SohamTilekar** [python-pillow/Pillow#7646] [Q&A] [answered]
I want to make the Text image Using this Library.
I Know How to use the Draw text But I want to Dynamically Adjust the Image Width & height on the Basis of the text and align the text in the center.

My old Code:
```python
draw = ImageDraw.Draw(img)
# font = ImageFont.truetype(<font-file>, <font-size>)
font = ImageFont.truetype(size=16)
# draw.text((x, y),"Sample Text",(r,g,b))
draw.text((0, 0),"Sample Text",(255,255,255),font=font)
img.save('sample-out.jpg')
```

**Comment by radarhere** [comment, marked as answer]
To adjust the size of the image to fit the text, you can
```python
from PIL import Image, ImageFont, ImageDraw
font = ImageFont.load_default(size=16)
text = "Sample Text"
w, h = font.getbbox(text)[2:]
img = Image.new("RGB", (w, h))
draw = ImageDraw.Draw(img)
draw.text((0, 0), text, (255,255,255), font=font)
img.save('sample-out.jpg')
```
![sample-out](https://github.com/python-pillow/Pillow/assets/3112309/6ca6a00e-45a8-44cc-a145-815204070cb4)

However, you've also said you'd like the text to be aligned to the center. That doesn't mean anything if the image exactly fits the width of a single line of text, so I presume you're dealing with multiple lines of text?

```python
from PIL import Image, ImageFont, ImageDraw
font = ImageFont.load_default(size=16)
text = "Sample\nText"
img = Image.new("RGB", (0, 0))
draw = ImageDraw.Draw(img)
w, h = draw.multiline_textbbox((0, 0), text, font)[2:]
img = Image.new("RGB", (w, h))
draw = ImageDraw.Draw(img)
draw.text((0, 0), text, (255,255,255), font=font, align="center")
img.save('sample-out.jpg')
```
![sample-out](https://github.com/python-pillow/Pillow/assets/3112309/20b6ff8a-8797-4f2b-880e-2a8f93a66963)

**Reply by SohamTilekar** [reply]
Thanks I Needed it for Developing My Module Named [Vidiopy](https://github.com/SohamTilekar/vidiopy) Would You Like to Help Me on that project.

**Reply by radarhere** [reply]
Helping Pillow is enough for me, thanks.
